src/store/socket/socket.ts

```diff
@@ -22,26 +19,31 @@ export const initialState: SocketState = {
   connected: false
 };
 
-type SocketAction = StompConnected | StompDisconnected | StompMessage;
-
-export const reducer = (state: SocketState = initialState, action: SocketAction): SocketState => {
-  switch (action.type) {
-    case STOMP_CONNECTED:
-      return {
-        ...state,
-        connected: true
-      };
-    case STOMP_CLOSED:
-      return {
-        ...state,
-        connected: false
-      };
-    case STOMP_MESSAGE:
-      return {
-        ...state,
-        messages: [...state.messages, action.payload]
-      };
-    default:
-      return state;
+const socketSlice = createSlice({
+  name: 'chess/socket',
+  initialState,
+  reducers: {},
+  extraReducers: builder => {
```

Comment:
try using the other syntax here

Reply:
this syntax is recommended with TypeScript
https://redux-toolkit.js.org/api/createSlice#the-extrareducers-map-object-notation